### PR TITLE
Skeleton states Case 3 Public Subscription

### PR DESCRIPTION
**Pull request by gsvr**
Based on [v7b flow pdf](https://drive.google.com/a/praekeltconsulting.com/file/d/0B7R1YAPd1mwcOXQ3NnRWR3EyRTQ/edit?usp=sharing)

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review

File: README.rst [hunk 1]
```diff
@@ -1,19 +1,19 @@
-Go Hack App for mHealth Connectathon
-====================================
+go-jsbox-skeleton
```
Comment (imsickofmaps) [inline README.rst:1]: Restore a proper README

File: go-app.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    var ChoiceState = vumigo.states.ChoiceState;
+    var EndState = vumigo.states.EndState;
+    var FreeText = vumigo.states.FreeText;
+
+    var GoNDOH = App.extend(function(self) {
+        App.call(self, 'states:start');
+        var $ = self.$;
+
+        self.states.add('states:start', function(name) {
+            return new ChoiceState(name, {
```
Comment (imsickofmaps) [inline go-app.js:17]: Please use the `LanguageChoice` state: http://vumi-jssandbox-toolkit.readthedocs.org/en/release-0.2.x/states.html#LanguageChoice

File: go-app.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                    'pregnant mothers. Are you or do you suspect that you ' +
+                    'are pregnant?'),
+
+                choices: [
+                    new Choice('states:id_type', $('Yes')),
+                    new Choice('states:end_not_pregnant', $('No')),
+                ],
+
+                next: function(choice) {
+                    return choice.value;
```
Comment (imsickofmaps) [inline go-app.js:48]: put the logic for choosing next state based on the choice.value here - and in other states

File: LICENSE [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE 
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+POSSIBILITY OF SUCH DAMAGE.
```
Comment (imsickofmaps) [inline LICENSE:28]: just add a line break

File: go-app.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                }
+            });
+        });
+
+        self.states.add('states:birth_month', function(name) {
+            return new ChoiceState(name, {
+                question: $('Please enter the month that you were born.'),
+
+                choices: [
+                    new Choice('states:birth_day', $('Jan')),
```
Comment (imsickofmaps) [inline go-app.js:138]: return the two digit month so we can build an ISO date string later. Also make a self.make_month_choices with a `start` and `limit` param.

File: go-app.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+                question: $('Please enter the day that you were born ' +
+                    '(eg 14).'),
+
+                next: function() {
+                    return 'states:end_success';
+                }
+            });
+        });
+
+        // text shortened - too many characters
```
Comment (imsickofmaps) [inline go-app.js:169]: Please email me the original and your shortened so I can log a JIRA ticket for approval

**Comment by imsickofmaps** [comment]
@gsvr - as discussed, please make the changes above to the multi-app branch you're working on and cross check with this list after you've PR'd it.

**Comment by smn** [comment]
When should I expect to be able to review this?

**Pull request merged**